### ISSUE TITLE
viz::Viz3d window closes automatically after thread being ended

### ISSUE THREAD

**Issue by JacobZhu** [Itseez/opencv#6378]
### Please state the information for your system
- OpenCV version: 3.1
- Host OS: Windows 10

### In which part of the OpenCV library you got the issue?
Examples:
- viz
- viz::Viz3d window closes automatically after thread being ended

### Additional description
I want to visualize a gradually growing point cloud using viz::Viz3d, and I've tried to show and update a viz::Viz3d window from within a thread, it runs just OK, but when the thread ends the viz::Viz3d window also disappears, which I think is abnormal, I expected the viz::Viz3d window to stay open so I can check out the overall quality of the point cloud when it is completed.

### Code example to reproduce the issue / Steps to reproduce the issue

the basic structure of my codes is shown below (kinda like a pseudo-code):

viz::Viz3d wnd3d("3D Window");

UINT Thread(LPVOID param)
{
	for (int i = 0; i < nImgs; ++i)
	{
		Mat XYZs = GeneratePointCloud(vImgs[i]);

		viz::WCloud cld(XYZs);

		wnd3d.showWidget("cloud", cld);

		wnd3d.spinOnce();
	}

	return TRUE;
}

void OnClickMenu()
{
	// TODO: Add your command handler code here
	AfxBeginThread(Thread, this, THREAD_PRIORITY_NORMAL);
}

**Comment by mshabunin** [comment]
Your widgets are stored in local thread variable. Did you try to put them to main thread?

**Comment by JacobZhu** [comment]
@mshabunin Yes, my widget viz::WCloud is stored in a local thread variable, but I thought it will be OK as long as my viz::Viz3d object is a global variable, and I also thought that once wnd3d.showWidget("cloud", cld) is executed I will not need a viz::WCloud object anymore.

Anyway, I will try put the viz::WCloud object to main thread and see how it goes.

**Comment by JacobZhu** [comment]
@mshabunin I have tried to put viz::WCloud object to main thread, but no, it still does not work as long as viz::Viz3d window is called and updated from within a thread.
